src/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.cs

```diff
@@ -1202,18 +1202,9 @@ internal Object InternalGetValue(String name, Object defaultValue, bool doNotExp
                         // make sure the string is null terminated before processing the data
                         if (blob.Length > 0 && blob[blob.Length - 1] != (char)0)
                         {
-                            try
-                            {
-                                char[] newBlob = new char[checked(blob.Length + 1)];
-                                Array.Copy(blob, 0, newBlob, 0, blob.Length);
-                                newBlob[newBlob.Length - 1] = (char)0;
-                                blob = newBlob;
-                            }
-                            catch (OverflowException e)
-                            {
-                                throw new IOException(SR.Arg_RegGetOverflowBug, e);
-                            }
-                            blob[blob.Length - 1] = (char)0;
+                            char[] newBlob = new char[blob.Length + 1];
+                            Array.Copy(blob, 0, newBlob, 0, blob.Length);
+                            blob = newBlob;
```

Comment:
I second @stephentoub for adding the test. 
Array.Resize is good idea too but it'll add a couple of checks. I still prefer using Array.Resize as it is explicit and simpler. I don't think the extra checks will make much difference.